agentic/src/agent_construct.rs

```diff
@@ -0,0 +1,3 @@
+pub trait AgentConstruct: Sized {
+    fn construct_from_params(params: Vec<golem_wasm_rpc::WitValue>, agent_id: String) -> Self;
+}
```

Comment:
As a user, I don't think it hurts, makes things all compile driven. 
If we are implementing AssistantAgent for `T`, compiler will ask you to have `AgentConstruct` instance for T which is available through `derive[AgentConstruct]` . Any other way is opening up lots of random possibilities, and getting it out for control

Reply:
I will have another PR very soon to have it all wit-value changed to data-value